config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/FolderLoader.java

```diff
@@ -40,8 +42,10 @@ public void load(Bundle bundle, Element element) {
             parentFolder = ServiceAndPolicyLoaderUtil.getFolder(bundle, parentFolderID);
         }
 
-        if (CharacterBlacklistUtil.containsInvalidCharacter(name)) {
-            throw new BundleLoadException("Folder name contains invalid characters: " + name);
+        try {
+            name = CharacterBlacklistUtil.encodeName(name);
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "unable to encode folder name " + name);
```

Comment:
@uppoju , during the loading time as well, when an exception comes we should fail the build